### PR TITLE
refactor: upgraded Nextjs from v10 to v12.

### PR DESCRIPTION
**Pull request by zhyd1997**
## Description

Upgraded Nextjs from v10 to v12 directly.

## Related issue

Fixes: https://github.com/Kwenta/kwenta/issues/741
Fixes: https://github.com/Kwenta/Kwenta/issues/596

## Motivation and Context

<del>replacing `optimized-images-loader` and `react-optimized-image` with `next/image` ?</del>
replaced `optimized-images-loader` and `react-optimized-image` with `next-optimized-images` (started at commit https://github.com/Kwenta/kwenta/pull/801/commits/bd86307bbccf15c105e08265b1c1f7ab7a272248).

now it's possible to use `next-transpile-module` to close issue #405.

## How Has This Been Tested?

For `next-optimized-images` integration, import all changed `svg` files into `DashboardContainer` directly and test if it works.

## Screenshots (if appropriate):

webpack v5

### CODE REVIEW

**Comment by brossetti1** [comment]
this one is relevant here 
https://github.com/Kwenta/kwenta/issues/596

looks like @zhyd1997 is handling this - upgrading the nextjs v12 and using `next/image` should solve some of the issues with `react-optimized-images` - we had a discussion to create one `Icon` component that loads svg using svgo so we can consolidate svg loading to one place and ditch next-optimized-image altogether. I am not sure where this stands currently but it might be worth looking into here OR adding before this upgrade goes through?

**Comment by zhyd1997** [comment]
@brossetti1 

Get it, it's easy to revert to commit https://github.com/Kwenta/kwenta/pull/801/commits/1834c8725b0bf18abab37d39833ce548ca91dc91 if we need adding before this upgrade.

**Comment by brossetti1** [comment]
@zhyd1997 ill talk Ralf and the other guys about this tomorrow and report back - great work on this!

**Comment by platschi** [comment]
Images for transaction confirmation looks a bit weird :

### How it should be

Screen Shot 2022-05-10 at 10 42 52

### How it looks like on this branch

Screen Shot 2022-05-10 at 10 44 06

**Comment by zhyd1997** [comment]
> Images for transaction notifications looks a bit weird :
> 
> ### How it should be
> Screen Shot 2022-05-10 at 10 42 52
> 
> ### How it looks like on this branch
> Screen Shot 2022-05-10 at 10 44 06

Seems that the icon size is different, I will have a look tomorrow.

**Review by brossetti1 — CHANGES_REQUESTED** [review]
this is looking really good so far. I have left some questions to be addressed and looks like youve already gone ahead and made some changes addressing some of the feedback, Ill wait for your response before moving forward, thanks @zhyd1997

File: next-env.d.ts
Comment (zhyd1997) [reply next-env.d.ts:3]: Nextjs automatically removed it when running `npm run dev` with current TS version.

Comment (zhyd1997) [reply next-env.d.ts:3]: You can try it if you added it manually.

File: package.json
Comment (zhyd1997) [reply package.json:58]: No worry about this, if there is no use for these packages, it will not affect the bundle size and app.

these packages are optional: https://github.com/cyrilwanner/next-optimized-images#optimization-packages

Comment (brossetti1) [reply package.json:58]: going to test some more things out today on this note.

Comment (zhyd1997) [reply package.json:58]: Why do we need to remove `next-optimize-image`, it did optimization for `svg` files.

Comment (zhyd1997) [reply package.json:58]: > I dont see these image libraries being used as a loader or called anywhere, are these providing some functionality with this upgrade? can you expand on this?
> 
> referring to these:
> 
> ```
> "image-trace-loader": "^1.0.2"
> "imagemin-mozjpeg": "^10.0.0"
> "jimp": "^0.16.1"
> "svg-sprite-loader": "^6.0.11",
> "responsive-loader": "^2.3.0",
> ```

https://github.com/zhyd1997/kwenta/pull/2

**Review by brossetti1 — DISMISSED** [review]
had some discussions with @zhyd1997 and looked this over - everything looks good from my side - tested a few different scenarios.

still it would be good to get another pair of eyes on this one before a merge.

**Comment by zhyd1997** [comment]
Yes, need test by team so that it will not break the whole app.

I will test again today (UTC+8).

File: components/MarketClosureIcon/MarketClosureIcon.tsx
Comment (koredefashokun) [inline components/MarketClosureIcon/MarketClosureIcon.tsx:26]: I think we can safely either delete this line or uncomment it and remove the className prop on Line 29

File: package.json
Comment (koredefashokun) [inline package.json:59]: Maybe we can pin this to a specific version.

File: package.json
Comment (koredefashokun) [inline package.json:135]: We might also want to pin they versions for `typescript` and `@svgr/webpack`.

File: sections/exchange/BasicSwap/BasicSwap.tsx
Comment (koredefashokun) [inline sections/exchange/BasicSwap/BasicSwap.tsx:15]: I think I might be missing something (maybe a TS issue), but is there a reason we aren't just doing `<ArrowIcon className="arrow" />` directly?

File: sections/exchange/TradeCard/Charts/common/ChartTypeIconsToggle.tsx
Comment (koredefashokun) [inline sections/exchange/TradeCard/Charts/common/ChartTypeIconsToggle.tsx:60]: I think you can put the filter value in a variable and pass it as props to remove the need for duplication.

File: sections/futures/Orders/Orders.tsx
Comment (koredefashokun) [inline sections/futures/Orders/Orders.tsx:283]: Not sure how this particular one works exactly, but I think the check is redundant, since this is already the PendingIcon. You should probably be able to set the color to yellow without checking.

File: sections/homepage/ChartBanner/ChartBanner.tsx
Comment (koredefashokun) [inline sections/homepage/ChartBanner/ChartBanner.tsx:26]: Long shot, but you could probably create a `Webp` component. That besically takes in the `ChartBanner` and generates the rest. I think there are some other places in this PR where you could reuse it.

**Review by koredefashokun — COMMENTED** [review]
Generally, the changes look good. Left a couple of comments mostly around cleaning and reusing code.

File: sections/exchange/TradeCard/Charts/common/ChartTypeIconsToggle.tsx
Comment (zhyd1997) [reply sections/exchange/TradeCard/Charts/common/ChartTypeIconsToggle.tsx]: @koredefashokun You leveled up my TS skills :thumbsup:

**Review by koredefashokun — DISMISSED** [review]

**Comment by koredefashokun** [comment]
Approved. Code changes look great. Thanks @zhyd1997

**Review by brossetti1 — APPROVED** [review]
this looks good to me, great job @zhyd1997!!

**Review by platschi — APPROVED** [review]

**Pull request merged**
